server/tls/tlscertificate_retriever_linux.go

```diff
@@ -0,0 +1,112 @@
+// +build linux
+
+// Copyright 2020 Microsoft. All rights reserved.
+
+package tls
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+)
+
+const (
+	CertLabel       = "CERTIFICATE"
+	PrivateKeyLabel = "PRIVATE KEY"
+)
+
+type linuxTlsCertificateRetriever struct {
+	pemBlock []*pem.Block
+	settings TlsSettings
+}
+
+// GetCertificate Returns the certificate associated with the pem
+func (fcert *linuxTlsCertificateRetriever) GetCertificate() (*x509.Certificate, error) {
+	for _, block := range fcert.pemBlock {
+		if block.Type == CertLabel {
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to parse certificate at location %s with error %+v", fcert.settings.TLSCertificatePath, err)
+			}
+			if !cert.IsCA {
+				return cert, nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("No Certificate block found")
+}
+
+// GetPrivateKey Returns the private key associated with the pem
+func (fcert *linuxTlsCertificateRetriever) GetPrivateKey() (crypto.PrivateKey, error) {
+	for _, block := range fcert.pemBlock {
+		if block.Type == PrivateKeyLabel {
+			pk, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("Could not parse private key %+v", err)
+			}
+			return pk, nil
+		}
+	}
+	return nil, fmt.Errorf("No private key found in certificate bundle located at %s", fcert.settings.TLSCertificatePath)
+}
+
+// ReadFile reads a from disk
+func (fcert *linuxTlsCertificateRetriever) ReadFile() ([]byte,error) {
+	content, err := ioutil.ReadFile(fcert.settings.TLSCertificatePath)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading file from path %s with error: %+v ", fcert.settings.TLSCertificatePath, err)
+	}
+	return content, nil
+}
+
+// Parses a file to PEM format
+func (fcert *linuxTlsCertificateRetriever) ParsePEMFile(content []byte) (error) {
+	pemBlocks := make([]*pem.Block, 0)
+
+	var pemBlock *pem.Block
+	nextPemBlock := content
+
+	for {
+		pemBlock, nextPemBlock = pem.Decode(nextPemBlock)
+
+		if pemBlock == nil {
+			break
+		}
+		pemBlocks = append(pemBlocks, pemBlock)
+	}
+
+	if len(pemBlocks) < 2 {
+		return fmt.Errorf("Invalid PEM format located at %s", fcert.settings.TLSCertificatePath)
+	}
+
+	fcert.pemBlock = pemBlocks
+	return nil
+}
+
+// ReadFile reads a from disk
+// linux uses file permissions and does not need to encrypt the file
+func (fcert *linuxTlsCertificateRetriever) Decrypt(content []byte) (string, error) {
```

Comment:
Once the interface is shortened, this method won't be needed

Reply:
Removed